sdk/storage/Azure.Storage.Common/src/Shared/ClientsideEncryption/ClientSideEncryptionVersionExtensions.cs

```diff
@@ -9,38 +11,62 @@ public static class ClientSideEncryptionVersionString
         {
             public const string V1_0 = "1.0";
             public const string V2_0 = "2.0";
+            public const string V2_1 = "2.1";
         }
 
-        public static string Serialize(this ClientSideEncryptionVersion version)
+        public static string Serialize(this ClientSideEncryptionVersionInternal version)
         {
             switch (version)
             {
 #pragma warning disable CS0618 // obsolete
-                case ClientSideEncryptionVersion.V1_0:
+                case ClientSideEncryptionVersionInternal.V1_0:
                     return ClientSideEncryptionVersionString.V1_0;
 #pragma warning restore CS0618 // obsolete
-                case ClientSideEncryptionVersion.V2_0:
+                case ClientSideEncryptionVersionInternal.V2_0:
                     return ClientSideEncryptionVersionString.V2_0;
+                case ClientSideEncryptionVersionInternal.V2_1:
+                    return ClientSideEncryptionVersionString.V2_1;
                 default:
                     // sanity check; serialize is in this file to make it easy to add the serialization cases
                     throw Errors.ClientSideEncryption.ClientSideEncryptionVersionNotSupported();
             }
         }
 
-        public static ClientSideEncryptionVersion ToClientSideEncryptionVersion(this string versionString)
+        public static ClientSideEncryptionVersionInternal ToClientSideEncryptionVersion(this string versionString)
         {
             switch (versionString)
             {
 #pragma warning disable CS0618 // obsolete
                 case ClientSideEncryptionVersionString.V1_0:
-                    return ClientSideEncryptionVersion.V1_0;
+                    return ClientSideEncryptionVersionInternal.V1_0;
 #pragma warning restore CS0618 // obsolete
                 case ClientSideEncryptionVersionString.V2_0:
-                    return ClientSideEncryptionVersion.V2_0;
+                    return ClientSideEncryptionVersionInternal.V2_0;
+                case ClientSideEncryptionVersionString.V2_1:
+                    return ClientSideEncryptionVersionInternal.V2_1;
                 default:
                     // This library doesn't support the stated encryption version
                     throw Errors.ClientSideEncryption.ClientSideEncryptionVersionNotSupported(versionString);
             }
         }
     }
+
+    internal enum ClientSideEncryptionVersionInternal
+    {
+        /// <summary>
+        /// 1.0.
+        /// </summary>
+        [Obsolete("This version is considered insecure. Applications are encouraged to migrate to version 2.0 or to one of Azure Storage's server-side encryption solutions. See http://aka.ms/azstorageclientencryptionblog for more details.")]
```

Comment:
we don't need this in an internal enum. also removing it will probably let us get rid of all these warning disables